File: tareas/Tarea 13/RUBY/BusquedaEnVector.rb
#BUSQUEDA EN VECTOR

#Convertir a entero
def to_int(string)
    Integer(string)
rescue ArgumentError
    nil
end

#Contador
cont = 0

print "Tamaño del arreglo: "
lectura = gets.chomp

tamaño = to_int(lectura)

if (tamaño.instance_of? Integer) == true
    if tamaño < 0 or tamaño == 0
        puts "#{tamaño} no es un tamaño valido."
		puts "Presiona ENTER para salir."
		gets.chomp
        exit
    else
        vector = Array.new(tamaño)
        puts "El tamaño es de #{vector.size}"

        #Secuencia iterativa para indices en el vector
        for i in 0..vector.size-1 do
            print "Posicion[#{i}]: "
            pos = gets.chomp
            vector[i] = pos
        end

        #Busqueda del numero
        #Lectura
        print "Valor a buscar: "
        valor = gets.chomp

        #Recorrer vector
        for i in 0..vector.size-1 do
            if valor == vector[i]
                cont = cont + 1
            end
        end

        #Impresion
        puts "El numero #{valor} aparece #{cont} veces."
		puts "Presiona ENTER para salir."
		gets.chomp

    end

else
    puts "Valor invalido."
	puts "Presiona ENTER para salir."
	gets.chomp
end

File: tareas/Tarea 13/JAVA/BusquedaEnVector.java
//TAREA 13
//Busqueda en Vector

import java.util.Scanner;

public class BusquedaEnVector {
    public static void main (String[] args){
        //Contador
        int cont = 0;

        //Posicion
        int pos;

        //Variable de lectura de datos
        Scanner lect = new Scanner (System.in);

        //CREACION DE VECTOR
        int[] vector;

        System.out.print("Tamaño del arreglo: ");
        int tamaño = lect.nextInt();

        if (tamaño < 0){
            System.out.print("[-]No puede ser un tamaño negativo. Saliendo...");
            System.out.println("");
            System.exit(0);
        }

        else if (tamaño == 0){
            System.out.print("[-]No puede haber una cantidad igual a 0. Saliendo..");
            System.out.println("");
            System.exit(0);
        }

        vector = new int[tamaño];

        //Secuencia iterativa para indices en el vector
        for (int i=0;i<vector.length;i++){
            System.out.print("Posicion ["+ i + "]: ");
            pos = lect.nextInt();
            vector[i] = pos;
        }

        //BÚSQUEDA DEL NÚMERO
        //Lectura
        System.out.print("Valor a buscar: ");
        int valor = lect.nextInt();

        //Recorrer vector
        for (int i=0;i<vector.length;i++){
            if (valor == vector[i]){
                cont++;
            }
        }

        //Impresion
        System.out.print("El numero " + valor + " aparece " + cont + " veces.");
        //Salto de linea
        System.out.println("");

    }
}